### ISSUE TITLE
Remove leading '_' from all source file names

### ISSUE THREAD

**Issue by CIPop** [dotnet/corefx#3116]
Remove all leading underscores '_'  from all file names. These were previously indicating that the classes defined within them are internal.

Namespaces: System.IO, System.Net, System.Resources.ReaderWriter.

**Comment by davidsh** [comment]
Does this issue track whether to keep the underscores in the names and standardize across the rest of CoreFx for this pattern?

Or does this issue track whether to remove the underscore from the name of the file?

Might be better to re-title this issue to clarify what the end goal is beyond getting a discussion started.

**Comment by CIPop** [comment]
@joshfree This is not System.Net specific. 
@davidsh I've clarified the text in the issue description.

**Comment by Page-Not-Found** [comment]
I am interested to work on this. Is this still being tracked to be implemented?

**Comment by CIPop** [comment]
@Page-Not-Found Feel free to send out a PR at any time. Thanks for offering! 

If you do start work on this, please assign it to yourself and remove the "up-for-grabs" tag to ensure nobody else works on the same task. I can help with this part if you don't have the right permissions.

**Comment by akoeplinger** [comment]
@CIPop contributors don't have permission to change labels or being assigned issues, so you'd need to do that (i.e. apply the "grabbed by community" label).

**Comment by CIPop** [comment]
> contributors don't have permission to change labels or being assigned issues, so you'd need to do that (i.e. apply the "grabbed by community" label).

@akoeplinger Thanks. Done.

**Comment by Page-Not-Found** [comment]
FYI. I am working on this.

**Comment by Page-Not-Found** [comment]
What is the branch should i be using? Future branch right?

**Comment by CIPop** [comment]
No, the dotnet/master branch should be used for the PR.

**Comment by Page-Not-Found** [comment]
Okie Got you.

**Comment by Page-Not-Found** [comment]
There are files the class itself has leading "_" in them. Do we rename the class as well, I hope no since that would be breaking change and all the reference needs to be updated. 

Need some advice with the following 
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src\System.Console\src\System\IO\__Error.cs
src\System.IO.FileSystem.DriveInfo\src\System\IO\__Error.cs
src\System.IO.FileSystem\src\System\IO\__Error.cs
src\System.IO.Pipes\src\System\IO\__Error.cs
src\System.IO.UnmanagedMemoryStream\src\Common\__Error.cs
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

There is already file named as "SslStream.cs" with in the same folder. 
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src\System.Net.Security\src\System\Net\SecureProtocols\_SslStream.cs
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

**Comment by davidsh** [comment]
The class name and filename of the *.cs file should match.

In terms of conflict, like `src\System.Net.Security\src\System\Net\SecureProtocols\_SslStream.cs` which conflicts with a classname/filename of `SslStream`, just rename the class and filename to something like "SslStreaminternal" (for the classname) and `SslStreamInternal.cs` for the filename.

**Comment by Page-Not-Found** [comment]
Thanks. I am almost done with this changes. But Stuck up with the test cases failing, I am not able to figure why the file rename should fail. The Code builds just fine in the Visual Studio. 

Machine: WIn7
VS: Both 2013 / 2015
GitHub\corefx\src\System.Globalization.Extensions\tests\System.Globalization.Extensions.Tests.csproj" (BuildAndTest target(s)) -- FAILED.

The full log file is in this path 
https://gist.github.com/Page-Not-Found/71b5da330046f6436f85#file-msbuillog

Can some one help me with this?
Forked Branch: https://github.com/Page-Not-Found/corefx/tree/removeunderscore

**Comment by davidsh** [comment]
So, the failure is occuring in the System.Globalization.Extensions library which is unrelated to your "_" file rename changes?

I'm not sure what is going on with your dev machine and Win7.  I would suggest you submit your PR with your changes and we can then see how the CI machines handle things.  Perhaps this is just an issue related to your machine setup.

**Comment by davidsh** [comment]
Fixed via #4464
